expression/partition_pruner.go

```diff
@@ -123,9 +134,11 @@ func (p *hashPartitionPruner) tryEvalPartitionExpr(piExpr Expression) (val int64
 			val := p.constantMap[idx]
 			if val != nil {
 				pi.GetArgs()[0] = val
-				ret, _, err := pi.EvalInt(p.ctx, chunk.Row{})
+				ret, isNull, err := pi.EvalInt(p.ctx, chunk.Row{})
 				if err != nil {
 					return 0, false, false
+				} else if isNull {
+					return 0, true, true
 				}
 				return ret, true, false
```

Comment:
`return ret, true, isNull` is enough?

Reply:
yes, for expression like `(year(null) + 2)` or `year(null)`, MySQL and TiDB both produce `NULL`

Reply:
what I mean is that: 
we can remove line 140~141, and change line 143 to `return ret, true, isNull`